thedebian/elementor
# Add VideoPress as a video provider

## PR Checklist



## PR Type
What kind of change does this PR introduce?


## Summary

This PR can be summarized in the following changelog entry:

* Add VideoPress as a video widget source.
* Use the VideoPress player for self-hosted URLs using VideoPress.

## Description

This PR adds VideoPress as a new video source for the video widget.
It also makes sure the VideoPress player is used whenever a self-hosted URL is using VIdeoPress as backend, instead of including the MP4 URL.

## Test instructions
This PR can be tested by following these steps:

* Apply this PR
* Create a new page or post
* In the widget search bar, type "videopress"
* ✅ The video widget should be visible
* Insert a video widget
* Change the source to "VideoPress"
* ✅ The video `https://videopress.com/v/Ygmx4akX` should be loaded and displaying the VideoPress player
* Change the URL to `https://video.wordpress.com/embed/Ygmx4akX`
* ✅ The video should still be properly loaded
* Play with the controls
* ✅ The controls parameters should respond as expected
* For the next step, two possibilites:
    * On a site with an active VideoPress subscription
    * Or Install Jetpack VideoPress and choose the free plan and make sure VideoPress is enabled
* Go to your media library and upload a video
* Copy the video URL starting with "videos.files.wordpress.com" or "video.wordpress.com" or "videopress.com"
* Go back to your post/page or create a new one
* Insert a video widget and select "self-hosted"
* Select the video you uploaded on VideoPress from your media library
* ✅ The VideoPress player should be displayed instead of the default HTML video tag
* Select "external URL" and paste the URL you previously copied
* ✅ The VideoPress player should be displayed and the proper video should load.

## Quality assurance


Fixes #
